torchsde/_core/methods/euler_heun.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from .. import base_solver
+from ...settings import SDE_TYPES, NOISE_TYPES, LEVY_AREA_APPROXIMATIONS
+
+
+class EulerHeun(base_solver.BaseSDESolver):
+    weak_order = 1.0
+    sde_type = SDE_TYPES.stratonovich
+    noise_types = (NOISE_TYPES.additive, NOISE_TYPES.diagonal, NOISE_TYPES.general, NOISE_TYPES.scalar)
+    levy_area_approximations = LEVY_AREA_APPROXIMATIONS.all()
+
+    def __init__(self, sde, **kwargs):
+        if sde.noise_type == NOISE_TYPES.additive:
+            self.strong_order = 0.5
+        else:
+            self.strong_order = 1.0
+        super(EulerHeun, self).__init__(sde=sde, **kwargs)
+
+    def step(self, t0, t1, y0):
+        dt = t1 - t0
+        I_k = self.bm(t0, t1)
+
+        f_eval = self.sde.f(t0, y0)
+        g_prod_eval = self.sde.g_prod(t0, y0, I_k)
```

Comment:
It looks like we're standardising on a convention of calling these local variables `f` and `g_prod`. (Similarly for `g_prod_eval_prime `.) c.f. Heun.

Reply:
Done!